src/Tools/dotnet-monitor/Egress/Extension/ExtensionEgressProvider.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.Egress.Configuration;
+using Microsoft.Diagnostics.Tracing.Parsers.MicrosoftWindowsTCPIP;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Egress
+{
+    /// <summary>
+    /// Egress provider for egressing stream data to an Azure blob storage account.
+    /// </summary>
+    /// <remarks>
+    /// Blobs created through this provider will overwrite existing blobs if they have the same blob name.
+    /// </remarks>
+    internal partial class ExtensionEgressProvider :
+        EgressProvider<ExtensionEgressProviderOptions>
+    {
+        private readonly IEgressPropertiesProvider _propertyProvider;
+
+        public ExtensionEgressProvider(IEgressPropertiesProvider propertyProvider, ILogger<ExtensionEgressProvider> logger)
+            : base(logger)
+        {
+            _propertyProvider = propertyProvider;
+        }
+
+        public override async Task<string> EgressAsync(
+            string providerCategory,
+            string providerName,
+            ExtensionEgressProviderOptions options,
+            Func<Stream, CancellationToken, Task> action,
+            EgressArtifactSettings artifactSettings,
+            CancellationToken token)
+        {
+            Dictionary<string, string> props = new Dictionary<string, string>();
+            foreach (string key in _propertyProvider.GetKeys())
+            {
+                if (_propertyProvider.TryGetPropertyValue(key, out string val))
+                {
+                    props.Add(key, val);
+                }
+            }
+
+            ExtensionEgressPayload payload = new ExtensionEgressPayload()
+            {
+                Settings = artifactSettings,
+                Configuration = options,
+                Properties = props,
+                ProfileName = providerName,
+            };
+
+            string startingPayload = JsonSerializer.Serialize<ExtensionEgressPayload>(payload);
+
+            // [TODO] add a new service to dynamically find these extensions
+            const string extensionProcessPath = "MyPathTo\\Extension.exe";
+            ProcessStartInfo pStart = new ProcessStartInfo()
+            {
+                FileName = extensionProcessPath,
```

Comment:
I would be explicit about parameters such as CreateNoWindow and UseShellExecute even if they have reasonable defaults on net core.